Microsoft.Azure.CosmosRepository/src/IRepository.cs

```diff
@@ -227,6 +227,29 @@ public interface IRepository<TItem> where TItem : IItem
         ValueTask<bool> ExistsAsync(Expression<Func<TItem, bool>> predicate,
             CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Queries cosmos DB to obtain the count of items.
+        /// </summary>
+        /// <remarks>
+        /// This queries the total number of documents in the container.
+        /// </remarks>
+        /// <param name="cancellationToken">The cancellation token to use when making asynchronous operations.</param>
+        /// <returns>A <see cref="ValueTask"/> representing the asynchronous count operation.</returns>
+        ValueTask<int> CountAsync(CancellationToken cancellationToken = default);

```

Comment:
What was your intention behind this API, count of the documents in the container of the given type? Just wondering if we need this or the below API could be made to do this by the user of the library if that makes sense?

Reply:
I went both ways on this. I spent a lot of time looking at my personal work and when I needed the count of a certain type. Most of the times, I used a filter to get a count of certain documents, but sometimes I did want the total count of one document type.

But yes this all could be achieved with the expression API such as - `CountAsync(item => true)` or similar. However, it also seems to me that code would be messier than `CountAsync()` in someone's implementation. That's why I left it in.

But now that I'm thinking about it, we could make an extension method to provide a no predicate option. I'm fine with any of the 3 options: leave as is, remove the call without the predicate, or add an extension method that hides the predicate.

Reply:
I like extension methods my only hesitation is testability as you can't mock extension methods. I think since each `IRepository<T>` is scoped to the specific `IItem` then it makes sense that this would only count those `IItem`s in a container. 

What do you think @IEvangelist ?

Reply:
So I believe we are saying to leave both methods?

Reply:
Yes I think so